File: ANEFreeInIty_AI/AIBackEnd/Data/Entity/Vector2D.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AIBackEnd.Data.Entity
{
    [Table("vector2d")]
    public class Vector2D
    {
        [Key]
        [Required(ErrorMessage = "Vector ID is required")]
        [Column("id")]
        public int Id { get; set; }

        [Column("x_quadinate_val")]
        public double X { get; set; }

        [Column("y_quadinate_val")]
        public double Y { get; set; }

        [Column("magnitude")]
        public double Magnitude { get; set; }

        [Column("angle_with_x_axis_deg")]
        public double AngleWithXAxisDeg { get; set; }

        [Required(ErrorMessage = "Description is required")]
        [Column("v_description")]
        public string Description { get; set; }

        [Required(ErrorMessage = "Name is required")]
        [Column("v_name")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Is Cartesian is required")]
        [Column("is_cartesian")]
        public bool IsCartesian { get; set; }
    }
}

File: FrontEnd/AIFrontEND/src/app/Model/Vector2D.ts
export class Vector2D {
    id : number | null | undefined;
       
    x : number | null | undefined;

    y : number | null | undefined;

    magnitude : number | null | undefined;

    angleWithXAxisDeg : number | null | undefined;

    isCartesian : boolean | null | undefined = true;

    description : string | null | undefined = "";

    name : string | null | undefined = "";
}

export interface SMalVector2D {
    vector: Vector2D;
    sVal: number;
}

export interface LinearCombVector2D {
    vectors: Vector2D[];
    xRange: number;
    yRange: number;
    scale: number | null;
}